### ISSUE TITLE
IRCUser shouldn't send topic if there isn't one.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2204]
|<img alt="colin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| colin reported|
|-|-|
|Trac ID|trac#2204|
|Type|defect|
|Created|2006-10-31 21:25:51Z|
|Branch|https://github.com/twisted/twisted/tree/not-send-topic-2204|

Check if there is a topic before sending data which confuses irssi. 

```
    def _sendTopic(self, group):
        topic = group.meta.get("topic")
        author = group.meta.get("topic_author") or "<noone>"
        date = group.meta.get("topic_date", 0)
        if topic:
            self.topic(self.name, '#' + group.name, topic)
            self.topicAuthor(self.name, '#' + group.name, author, date)
```

<details><summary>Searchable metadata</summary>

```
trac-id__2204 2204
type__defect defect
reporter__colin colin
priority__normal normal
milestone__ 
branch__branches_not_send_topic_2204 branches/not-send-topic-2204
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1162329951000000 1162329951000000
changetime__1222120499000000 1222120499000000
version__None None
owner__ 
cc__colin@...
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [24718]) Branching to 'not-send-topic-2204'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

My version of irssi doesn't seem to care if the topic is sent or not.  But I guess this can only be an improvement.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

 * Some flakes
```
twisted/words/test/test_service.py:8: 'implements' imported but unused
twisted/words/test/test_service.py:14: 'iwords' imported but unused
```
 * Please update copyright notice of both files
 * test_service contains a useless future import

Please merge once fixed.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [24907]) Update copyright date, add a module docstring for the test module, remove unused imports and obsolete future imports

refs [#2204](https://github.com/twisted/twisted/issues/2204)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [24908]) Merge not-send-topic-2204

Author: exarkun
Reviewer: therve
Fixes: [#2204](https://github.com/twisted/twisted/issues/2204)

Change the Twisted Words IRC server so that no topic message
is sent for a group which has no topic.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
